### PR TITLE
Added Fence API and fences in all wood types

### PR DESCRIPTION
**Pull request by LNJplus**
Just the documentation is missing.
But I can do it.

And I hope that the itemstrings and descriptions are all right.

Sorry I forgot the craftings. I'll do them now.

### CODE REVIEW

**Comment by BlockMen** [comment]
And how they should be crafted?

**Comment by LNJplus** [comment]
stick planks stick
stick planks stick

File: mods/default/nodes/fences.lua [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+			{'group:stick', def.craftitem, 'group:stick'},
+			{'group:stick', def.craftitem, 'group:stick'},
+		}
+	})
+end
+
+default.register_fence("default:fence_wood", {
+	description = "Wooden Fence",
+	texture = "default_wood.png",
+	craftitem = "default:wood"
```
Comment (BlockMen) [inline mods/default/nodes/fences.lua]: I think material would be better than craftitem

**Comment by BlockMen** [comment]
And keep the recipe for default:fence_wood plz (for backwards compatiblity)

**Comment by LNJplus** [comment]
Ok. Is all right now?

**Comment by LNJplus** [comment]
I tested my code and found an error but I don't know what I'm doing wrong:
```
========== ERROR FROM LUA ===========
ERROR[main]: Failed to load and run script from 
ERROR[main]: /home/lnj/.minetest/games/minetest_next/mods/default/init.lua:
ERROR[main]: ...netest/games/minetest_next/mods/default/nodes/fences.lua:38: '<eof>' expected near 'end'
ERROR[main]: stack traceback:
ERROR[main]: 	[C]: in function 'dofile'
ERROR[main]: 	...lnj/.minetest/games/minetest_next/mods/default/nodes.lua:52: in main chunk
ERROR[main]: 	[C]: in function 'dofile'
ERROR[main]: 	.../lnj/.minetest/games/minetest_next/mods/default/init.lua:36: in main chunk
ERROR[main]: ======= END OF ERROR FROM LUA ========```

File: mods/default/nodes/fences.lua [hunk 2]
```diff
@@ -0,0 +1,61 @@
+default.register_fence(name, def)
```
Comment (BlockMen) [inline mods/default/nodes/fences.lua]: it has to be ```function default.register_fence(name, def)```

**Comment by LNJplus** [comment]
Thanks!

**Comment by LNJplus** [comment]
First Screenshot:

**Comment by LNJplus** [comment]
I've added the documentation to game_api.txt.

File: game_api.txt [hunk 3]
```diff
@@ -249,6 +249,31 @@ xpanes.register_pane(subname, def)
...
+ -> itemstring: used for nodename and crafting recipes.
+ -> def: See [#Fence deinition]
+
+#Fence definition
+-----------------
+{
+	description = "Wooden Fence",
+	^ The name / description of the fence
+	texture = "default_wood.png",
+	^ The texture for the node. It is used for the inventory texture too, but with some tricks
```
Comment (BlockMen) [inline game_api.txt]: just a note that it is used for both is enough. It no back magic ;)

File: mods/default/nodes/fences.lua [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+			type = "fixed",
+			fixed = {-1/7, -1/2, -1/7, 1/7, 1/2, 1/7},
+		},
+		groups = def.groups,
+		sounds = def.sounds,
+	})
+	
+	
+	minetest.register_craft({
+		output = itemstring .. ' 4',
```
Comment (BlockMen) [inline mods/default/nodes/fences.lua]: Maybe 2 instead of 4?

File: game_api.txt [hunk 5]
```diff
@@ -249,6 +249,31 @@ xpanes.register_pane(subname, def)
 	^ Recipe field only
 }
 
+Fence API
+---------
+Creates fences that automatically connect to each other.
```
Comment (BlockMen) [inline game_api.txt]: Better: "Creates fence of given material"

File: game_api.txt [hunk 6]
```diff
@@ -249,6 +249,31 @@ xpanes.register_pane(subname, def)
...
+#Fence definition
+-----------------
+{
+	description = "Wooden Fence",
+	^ The name / description of the fence
+	texture = "default_wood.png",
+	^ The texture for the node. It is used for the inventory texture too, but with some tricks
+	  and the texture default_fence_overlay.png.
+	material = "default:wood",
+	^ The material wich you need to craft the fence.
```
Comment (BlockMen) [inline game_api.txt]: which, not wich

**Comment by BlockMen** [comment]
Beside the added comments you need to restore the "old" recipe for default:fence_wood.
Otherwise ok now.

File: mods/default/nodes/fences.lua (hunk 4 above)
Comment (LNJplus) [inline mods/default/nodes/fences.lua]: In the old recipe you get 1 fence for 3 sticks.
In the new recipe too:
1 plank = 4 sticks
4 (sticks) + 2 (planks) * 4 = 12
12 / 3 = 4 (fences)
I think 4 is good, but if you want I can change that.

**Comment by LNJplus** [comment]
What do you mean exactly? The old recipe in the crafting.lua exist, if you mean that.
https://github.com/LNJplus/minetest_next/blob/fence/mods/default/crafting.lua

File: mods/default/nodes/fences.lua (hunk 4 above)
Comment (BlockMen) [inline mods/default/nodes/fences.lua]: Hmm, ok. convinced

**Comment by BlockMen** [comment]
oh, sorry. Mix that up somehow. Could you remove the "tricks" part in game_api.txt and squash it so i can merge?

**Comment by LNJplus** [comment]
Is it ok now?

**Comment by BlockMen** [comment]
Yes, meged. b25477b3151bce34dfa66607843c14862cabbf90

**Pull request closed without merging**